### PR TITLE
crunchy-cli 3.0.3 (new formula)

### PR DESCRIPTION
**Pull request by kennedy**
Command-line downloader for Crunchyroll

depends on `ffmpeg` and `openssl@3`. `rust` for buiding

reviving previous PR attempt: https://github.com/Homebrew/homebrew-core/pull/139018/ https://github.com/Homebrew/homebrew-core/pull/140197
pre-req for https://github.com/Homebrew/homebrew-cask/pull/158759 if this is rejected

-----

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
reviving https://github.com/Homebrew/homebrew-core/pull/139018/
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
```
HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source crunchy-cli
==> Fetching crunchy-cli
==> Downloading https://github.com/crunchy-labs/crunchy-cli/archive/refs/tags/v3.0.3.tar.gz
Already downloaded: /Users/.../Library/Caches/Homebrew/downloads/a2ba919c9d35d1b86932d1759e897655fadbea291a3ea1cfe0ce0080fa000882--crunchy-cli-3.0.3.tar.gz
==> cargo install --no-default-features --features openssl-tls
==> Caveats
zsh completions have been installed to:
  /opt/homebrew/share/zsh/site-functions
==> Summary
🍺  /opt/homebrew/Cellar/crunchy-cli/3.0.3: 15 files, 5MB, built in 1 minute 18 seconds
==> Running `brew cleanup crunchy-cli`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
```
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
```
brew test crunchy-cli
==> Testing crunchy-cli
==> /opt/homebrew/Cellar/crunchy-cli/3.0.3/bin/crunchy-cli login --etp-rt invalid-token 2>&1
```
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?
```
brew audit --strict --new crunchy-cli
EOF
```
-----
Also did a style check
```
brew style crunchy-cli

1 file inspected, no offenses detected
```

### CODE REVIEW

**Comment by kennedy** [comment]
What an odd error:
```
A job completed hook has been configured by the self-hosted runner administrator
Run '/Users/brew/job_completed.sh'
  shell: /bin/bash --noprofile --norc -e -o pipefail {0}
Error: Process completed with exit code 7.
```

It ran perfectly but the post step died

File: Formula/c/crunchy-cli.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  head "https://github.com/crunchy-labs/crunchy-cli.git", branch: "master"
+
+  depends_on "rust" => :build
+  depends_on "ffmpeg"
+  depends_on "openssl@3"
+
+  on_linux do
+    depends_on "pkg-config" => :build
+  end
+
```
Comment (p-linnane) [inline Formula/c/crunchy-cli.rb:17]:
```suggestion
  depends_on "pkg-config" => :build
  depends_on "rust" => :build
  depends_on "ffmpeg"
  depends_on "openssl@3"

```

Comment (p-linnane) [reply Formula/c/crunchy-cli.rb:17]: `pkg-config` is in `rust`'s dep tree already, so we don't need to scope it to Linux.

File: Formula/c/crunchy-cli.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+class CrunchyCli < Formula
+  desc "Command-line downloader for Crunchyroll"
+  homepage "https://github.com/crunchy-labs/crunchy-cli"
+  url "https://github.com/crunchy-labs/crunchy-cli/archive/refs/tags/v3.0.3.tar.gz"
+  sha256 "9ee633b0de18b3cc2b6246c596b6b752a888e56203b8bc59db10dd7a8e4fa70a"
+  license "MIT"
+
```
Comment (p-linnane) [inline Formula/c/crunchy-cli.rb:7]:
```suggestion
```

File: Formula/c/crunchy-cli.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  on_linux do
+    depends_on "pkg-config" => :build
+  end
+
+  def install
+    system "cargo", "install", "--no-default-features", "--features", "openssl-tls", *std_cargo_args
+    man1.install Dir["target/release/manpages/*"]
+    bash_completion.install "target/release/completions/crunchy-cli.bash"
+    fish_completion.install "target/release/completions/crunchy-cli.fish"
+    zsh_completion.install "target/release/completions/_crunchy-cli"
```
Comment (p-linnane) [inline Formula/c/crunchy-cli.rb:23]: Does the executable have a function to generate these completions?

**Review by p-linnane — CHANGES_REQUESTED** [review]

File: Formula/c/crunchy-cli.rb (hunk 1 above)
Comment (kennedy) [reply Formula/c/crunchy-cli.rb:17]: Thanks for the detail I will accept your suggestion

File: Formula/c/crunchy-cli.rb (hunk 3 above)
Comment (kennedy) [reply Formula/c/crunchy-cli.rb:20]: I do not think so. I believe it is generated within the build process.

Comment (kennedy) [reply Formula/c/crunchy-cli.rb:19]: I will mention this upstream.

**Review by p-linnane — DISMISSED** [review]

**Pull request closed without merging**